docs/configuration/multiple-delivery-clients.md

```diff
@@ -1,36 +1,128 @@
-This SDK contains a default implementation of named clients relying on Autofac's [Named and Keyed Services](https://autofaccn.readthedocs.io/en/latest/advanced/keyed-services.html). If you wish to implement support for a DI container of your choice, jump to the [Extending named services support](#extending-named-services-support) section.
+> ⚠️ Mind [because of the issue with the named clients implementation](https://github.com/kontent-ai/delivery-sdk-net/issues/312) we decided to deprecate `AutofacServiceProviderFactory` use `MultipleDeliveryClientFactory` instead.
 
-# Using the default Autofac implementation
+Sometimes, it's handy to register multiple `IDeliveryClient`s with different configurations (e.g. while accessing different projects, accessing secured and non-secured data at once, or accessing preview and production data at the same time). In those cases, you can take advantage of multiple client registration using factory pattern.
 
-## Installing the NuGet packages
-Advanced registration scenarios are handled by the [Kontent.Ai.Delivery.Extensions.DependencyInjection](https://www.nuget.org/packages/Kontent.Ai.Delivery.Extensions.DependencyInjection) NuGet package. You'll need to install the following packages:
+If you wish to implement support for a DI container of your choice, jump to the [Extending named services support](#extending-named-services-support) section.
+
+## Using the default .NET implementation of the IMultipleDeliveryClientFactory
+
+### Installing the NuGet packages
+
+Advanced registration scenarios are handled by the [Kontent.Ai.Delivery.Extensions.DependencyInjection](https://www.nuget.org/packages/Kontent.Ai.Delivery.Extensions.DependencyInjection) NuGet package. You'll need to install it first:
+
+```sh
+Install-Package Kontent.Ai.Delivery.Extensions.DependencyInjection
+```
+
+### Registering the factory
+
+The SDK provides extension methods upon the `IServiceCollection` that allow registering delivery client factory builder. The builder is used to configure the factory and register the delivery client instances.
+
+```csharp
+public class Startup
+{
+    // ...
+
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddMultipleDeliveryClientFactory
+        (
+            factoryBuilder => factoryBuilder
+                .AddDeliveryClient
+                (
+                    "projectA",
+                    deliveryOptionBuilder => deliveryOptionBuilder
+                        .WithProjectId("<A_PROJECT_ID>")
+                        .UseProductionApi()
+                        .Build()
+                    optionalClientSetup =>
+                        optionalClientSetup.WithTypeProvider(new ProjectAProvider())
+                )
```

Comment:
It seems that
`.Build()`
is missed here

Reply:
ae78d43